web3/middleware/cache.py

```diff
@@ -1,24 +1,30 @@
 import functools
 import threading
 import time
-from typing import (
+from typing import (  # noqa: F401
```

Comment:
I would advocate putting the `Union` import inside the `if TYPE_CHECKING` block so that it is a bit clearer that it's the only one that the `F401` applies to.